src/ProjectConfiguration.ts

```diff
@@ -362,15 +362,21 @@ export function validateRequestedPermissions({
   });
 }
 
-export function validateBuildTarget({ buildTargets }: ProjectConfiguration) {
+export function validateBuildTarget(
+  { buildTargets }: ProjectConfiguration,
+  { hasNativeComponents }: { hasNativeComponents: boolean },
+) {
   const diagnostics = constrainedSetDiagnostics({
     actualValues: buildTargets,
     knownValues: knownBuildTargets,
     valueTypeNoun: 'build targets',
     notFoundIsFatal: true,
   });
 
-  if (buildTargets.length === 0) {
+  if (
+    (buildTargets === undefined || buildTargets.length === 0) &&
```

Comment:
i think that having `buildTargets` undefined as a valid option is in contrast with what our types are saying. We should change our typings too and adjust.

Reply:
It's only valid if there are native components, which is what this is checking for.

If it's undefined, the types are still correct, because the normalization function will default it to an empty array.